tests/GitDsc/GitDsc.tests.ps1

```diff
@@ -52,6 +52,22 @@ Describe 'GitDsc' {
         $finalState.FolderName | Should -Be $desiredState.FolderName
     }
 
+    It 'Clones a repository without checkout and file contents' {
+        $desiredState = @{
+            HttpsUrl      = 'https://github.com/microsoft/winget-dsc.git'
+            RootDirectory = $env:TEMP
+            ExtraArgs     = '--filter=blob:none --no-checkout'
+        }
+
+        Invoke-DscResource -Name GitClone -ModuleName GitDsc -Method Set -Property $desiredState
+
+        $finalState = Invoke-DscResource -Name GitClone -ModuleName GitDsc -Method Get -Property $desiredState
+        $finalState.HttpsUrl | Should -Be $desiredState.HttpsUrl
+        $finalState.Ensure | Should -Be 'Present'
+        $finalState.RootDirectory | Should -Be $desiredState.RootDirectory
+        $finalState.ExtraArgs | Should -Be $desiredState.ExtraArgs
```

Comment:
The check (`ExtraArgs`) isn’t needed because the Get operation doesn’t return that info, and it’s unlikely to be able to recover the initialization options anyway. Let’s just drop it.